Documentation/PackageCollections.md

```diff
@@ -0,0 +1,321 @@
+```
+
+#### Signed package collections
+
+Package collection publishers may sign a collection to protect its contents from being tampered with. If a collection is signed, SwiftPM will check that the 
+signature is valid before importing it and return an error if any of these fails:
+- The file's contents, signature excluded, must match what was used to generate the signature. In other words, this checks to see if the collection has been altered since it was signed.
+- The signing certificate must meet all the [requirements](#requirements-on-signing-certificate).
+
+```bash
+$ swift package-collection add https://www.example.com/bad-packages.json
+The collection's signature is invalid. If you would still like to add it please rerun 'add' with '--skip-signature-check'.
+```
+
+User may continue adding the collection despite of the error or opt to skip the signature check on a package collection by passing the `--skip-signature-check` flag:
+
+```bash
+$ swift package-collection add https://www.example.com/packages.json --skip-signature-check
+```
+
+For package collections hosted on the web, publishers may ask SwiftPM to [enforce the signature requirement](#protecting-package-collections). If a package collection is
+supposed to be signed but it isn't, user will see the following error message:
+
+```bash
+$ swift package-collection add https://www.example.com/bad-packages.json
+The collection is missing required signature, which means it might have been compromised.
+```
+
+Users should NOT add the package collection in this case.
+
+##### Trusted root certificates
+
+Since generating a collection signature requires a certificate, part of the signature check involves validating the certificate and its chain and making sure that the root certificate is trusted.
+
+On Apple platforms, all root certificates that come preinstalled with the OS are automatically trusted. Users may include additional certificates to trust by placing 
+them in the `~/.swiftpm/config/trust-root-certs` directory. 
+
+On non-Apple platforms, there are no trusted root certificates by default. Only those found in `~/.swiftpm/config/trust-root-certs` are trusted. This means that 
+the signature check will always fail unless the `trust-root-certs` directory is set up:
+
+```bash
+$ swift package-collection add https://www.example.com/packages.json
+The collection's signature cannot be verified due to missing configuration.
+```
+
+Users should obtain root certificates from collection publishers and save them to `~/.swiftpm/config/trust-root-certs`. The root certificates must be DER-encoded.
+
+#### Unsigned package collections
+
+Users will get an error when trying to add an unsigned package collection:
+
+```bash
+$ swift package-collection add https://www.example.com/packages.json
+The collection is not signed. If you would still like to add it please rerun 'add' with '--trust-unsigned'.
+```
+
+To continue user must confirm their trust by passing the `--trust-signed` flag:
```

Comment:
`--trust-unsigned`, right?

Reply:
Oops thanks. Fixed in https://github.com/apple/swift-package-manager/pull/3306/commits/0a51e58fd95b0718c9bc062b2df7cb3aff03bcfc